runtime/src/main/java/org/corfudb/runtime/CorfuRuntime.java

```diff
@@ -696,7 +696,17 @@ public synchronized void invalidateLayout() {
             // Don't create a new request for a layout if there is one pending.
             return;
         }
-        layout = fetchLayout(CFUtils.getUninterruptibly(layout).getLayoutServers());
+        List<String> lastKnownLayoutServers = layoutServers;
```

Comment:
Let us try to clean this up in a later version. This class is very unclean and exception handling is very unwieldy.